### PR TITLE
Add external dns advanced docs

### PR DESCRIPTION
**Pull request by mcharriere**
### What does this PR do?

Add external-dns advanced configuration docs

### What does it look like?

image

### Any background context you can provide?

https://github.com/giantswarm/roadmap/issues/411

### Have you maintained the front matter?

(Please bump the last_review_date in case this qualifies as a review for an entire page. Provide user_questions which should be answered in the page. Provide a meaningful description.)

### CODE REVIEW

File: src/content/advanced/external-dns/index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+linkTitle: External DNS
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:2]:
```suggestion
linkTitle: External DNS on AWS
```

File: src/content/advanced/external-dns/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+title: Advanced External DNS configuration
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:3]:
```suggestion
title: Advanced External DNS configuration for AWS
```

File: src/content/advanced/external-dns/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+description: How to customize advanced configuration of the External DNS service in your clusters.
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:4]:
```suggestion
description: How to configure the External DNS service in your clusters for use with AWS Route53.
```

File: src/content/advanced/external-dns/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  main:
+    parent: advanced
+last_review_date: 2022-12-06
+user_questions:
+  - How can I customize the External DNS service configuration?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-cabbage
+---
+
+## Integrate External DNS with AWS Route53 with access key
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:16]: This headline appears redundant. Normally the h1 headline (which is generated from the `title` attribute) should be enough as the article's headline.

File: src/content/advanced/external-dns/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+External DNS in Giant Swarm is configured to authenticate against the AWS using the method available on the cluster (KIAM or IRSA). But there are cases where this is no possible, for example, if you try to manage your DNS records in AWS Route53 from a cluster running on a different provider.
+
+There are 2 different possible configurations.
+
+### Use an existing secret (recommended)
+
+This method configures the App to mount the credentials file from an existing `external-dns-route53` secret.
+
+The secret must contain a file with the following format:
+```
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:27]:
```suggestion
The secret must contain a file with the following format:

```nohighlight
```

File: src/content/advanced/external-dns/index.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  - How can I customize the External DNS service configuration?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-cabbage
+---
+
+## Integrate External DNS with AWS Route53 with access key
+
+External DNS in Giant Swarm is configured to authenticate against the AWS using the method available on the cluster (KIAM or IRSA). But there are cases where this is no possible, for example, if you try to manage your DNS records in AWS Route53 from a cluster running on a different provider.
+
+There are 2 different possible configurations.
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:20]:
```suggestion
There are two different possible configurations.
```

File: src/content/advanced/external-dns/index.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  baseDomain: <domain>
+
+externalDNS:
+  aws_access_key_id: <key_id>
+  aws_secret_access_key: <secret>
+```
+
+__Warning:__
+
+This method is not recommended and will be deprecated in future versions.
```
Comment (marians) [inline src/content/advanced/external-dns/index.md:75]: I would move this as high up as possible.

**Comment by mcharriere** [comment]
@marians thanks for the review 🙏🏻 

> Two things:
> 
>     * As the entire article is AWS specific, I recommend to make the `linkTitle` and `title` include "AWS.

I plan to add more subsections with different use cases, so it wont be exclusively about AWS. Do you think we can keep the title as it is now or would it be better to be specific now and generalize later?

 
>     * The warning in the end seems important and should ideally be shown in the top of the article.

I'll move the warning right below the `Inject access key as values` title, as it is only relevant for that section.

**Comment by marians** [comment]
> I plan to add more subsections with different use cases, so it wont be exclusively about AWS. Do you think we can keep the title as it is now or would it be better to be specific now and generalize later?

You can adapt the titles then with those changes.

**Pull request merged**